assets/src/editor/events/dates-and-tickets-metabox/with-ticket-entities-for-filtered-date-entities.js

```diff
@@ -0,0 +1,35 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { getDateEntityIds } from '../dates-and-times/data/utils';
+import { intersection } from 'lodash';
+
+const DEFAULT_EMPTY_ARRAY = [];
+
+const withTicketEntitiesForFilteredDateEntities = createHigherOrderComponent(
+	withSelect( (
+		select,
+		{
+			filteredDateEntities = DEFAULT_EMPTY_ARRAY,
+			ticketEntities = DEFAULT_EMPTY_ARRAY,
+		}
+	) => {
+		const dateIds = getDateEntityIds( filteredDateEntities );
+		const { getRelationIdsForEntityRelation } = select( 'eventespresso/core' );
+		const allTicketEntities = [ ...ticketEntities ];
+		ticketEntities = ticketEntities.filter( ( ticketEntity ) => {
+			const relatedDateEntities = getRelationIdsForEntityRelation(
+				ticketEntity,
+				'datetime'
+			);
+			return relatedDateEntities.length > 0 &&
+				intersection( relatedDateEntities, dateIds ).length > 0;
+		} );
+		return {
+			allTicketEntities,
+			ticketEntities,
+		};
+	} ),
+	'withTicketEntitiesForFilteredDateEntities'
+);
+
+export default withTicketEntitiesForFilteredDateEntities;
```

Comment:
This is used anywhere we want `ticketEntities` to be those related to provided `filteredDateEntities` prop.  This too is an _interim_ refactoring step.  I think there's still a lot of unnecessary duplication of behaviour between all the filtering components etc so I hope in future refactors to simplify the flow a bit.